### DISCUSSION TITLE
Update lint configuration to use custom rules to improve VSCode plugin use.

### DISCUSSION THREAD

**Discussion by UniquePixels** [TheOdinProject/curriculum#27298] [Ideas]
Based on the discussion at https://github.com/TheOdinProject/.github/issues/23, I am opening a discussion here to discuss and research possible solutions.

Essentially, can we find a way to improve how we configure the linting setup so that the VSCode plugin also works for the file structure?

Essential links:
Plugin config: https://github.com/DavidAnson/vscode-markdownlint/tree/main?tab=readme-ov-file#configure
Possible solution: https://github.com/DavidAnson/vscode-markdownlint/tree/main?tab=readme-ov-file#markdownlintcustomrules

**Comment by mao-sz** [comment]
This is now a thing as of #28759
